### PR TITLE
Type warnings

### PR DESCRIPTION
**Pull request by daibhin**

### CODE REVIEW

File: ios/UnflowMetadataBridge.swift [hunk 1]
```diff
@@ -20,11 +20,20 @@ struct UnflowMetadataBridge {
...
+            if CFGetTypeID(numberValue) == CFBooleanGetTypeID() {
+                return numberValue.boolValue
+            } else {
+                return numberValue.doubleValue
+            }
         } else if let stringValue = value as? NSString {
-            return String(stringValue)
-        } else if let date = value as? NSDate {
-            return date as Date
+            let swiftString = String(stringValue)
```
Comment (daibhin) [inline ios/UnflowMetadataBridge.swift:35]: @SwiftyAlex I decided to actually write a parser to keep all the React weirdness in the same place. Could probably be more idiomatic if you care to take a look

File: ios/UnflowMetadataBridge.swift [hunk 2]
```diff
@@ -20,11 +20,20 @@ struct UnflowMetadataBridge {
         if let analyticsValue = value as? UnflowAnalyticsValue {
             return analyticsValue
         } else if let numberValue = value as? NSNumber {
-            return numberValue.doubleValue
+            if CFGetTypeID(numberValue) == CFBooleanGetTypeID() {
```
Comment (daibhin) [inline ios/UnflowMetadataBridge.swift:29]: @SwiftyAlex I noticed that we supported boolean types natively but not in the converter. This seems to work but not sure if it's correct

**Review by SwiftyAlex — DISMISSED** [review]

File: ios/UnflowMetadataBridge.swift [hunk 3]
```diff
@@ -9,6 +9,8 @@ import Foundation
 import UnflowUI
 
 struct UnflowMetadataBridge {
+    static let dateFormatter = ISO8601DateFormatter()
```
Comment (daibhin) [inline ios/UnflowMetadataBridge.swift]: Just tested and this doesn't parse the dates correctly. I think we need to add these two fomatting options
```
dateFormatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
```

I added them just above the `date` method but wasn't sure if there is a way to do it statically

File: ios/UnflowMetadataBridge.swift [hunk 4]
```diff
@@ -20,11 +22,19 @@ struct UnflowMetadataBridge {
...
+                return numberValue.boolValue
+            } else {
+                return numberValue.doubleValue
+            }
         } else if let stringValue = value as? NSString {
-            return String(stringValue)
-        } else if let date = value as? NSDate {
-            return date as Date
+            let swiftString = String(stringValue)
+            dateFormatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
```
Comment (SwiftyAlex) [inline ios/UnflowMetadataBridge.swift]: We can actually just set these where we instantiate the date formatter to not set them all the time ( not sure it matters )

something like this:

```
static var dateFormatter: DateFormatter = {
  let formatter = ISO8601...()
  formatter.options = ...
  return formatter
}()
```

Comment (daibhin) [reply ios/UnflowMetadataBridge.swift]: Ah that's exactly what I wanted. Didn't know how to do it in the constructor

**Review by SwiftyAlex — APPROVED** [review]
Splendid!

**Pull request merged**